.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,24 @@
+name: 'Setup Python'
+runs:
+  using: 'composite'
+  steps:
+  - name: Setup Python
+    uses: actions/setup-python@v4
+    with:
+      python-version: "3.11"
+    id: setup-python
+  - uses: snok/install-poetry@v1
+    with:
+      virtualenvs-create: true
+      virtualenvs-in-project: true
+      installer-parallel: true
+  - id: cached-poetry-dependencies
+    uses: actions/cache@v3
```

Comment:
I would try removing the cache until we get a full workflow running properly, and then we can add it back in.